src/textarea.rs

```diff
@@ -1162,6 +1293,31 @@ impl<'a> TextArea<'a> {
             hl.search(matches, self.search.style);
         }
 
+        // any selected text to highlight?
+        if self.select_start.get().is_some() {
+            let (start, end, _) = self.normalize_selection();
+
+            if start.0 <= row && end.0 >= row {
+                match (start.0 == row, end.0 == row) {
+                    (true, true) => {
+                        hl.select(start.1, end.1, self.select_style);
+                    }
+                    (true, false) => {
+                        // the +1 extends the highlight one beyond, to show
+                        // that the newline is included
+                        hl.select(start.1, line.len() + 1, self.select_style);
+                    }
+                    (false, true) => {
+                        if end.1 > 0 {
+                            hl.select(0, end.1, self.select_style);
+                        }
+                    }
+                    (false, false) => {
+                        hl.select(0, line.len() + 1, self.select_style);
+                    }
+                }
+            }
+        }
```

Comment:
Can you make this hunk into one method like `LineHighligher::cursor_line` or `LineHighlighter::line_number`? I'd like to keep this method smaller and separate the logic for selection into the new method.